onnx/onnx_pb.h

```diff
@@ -31,17 +26,22 @@
 // This is used in e.g. ONNX's protobuf files: when building the main library,
 // it is defined as ONNX_EXPORT to fix a Windows global-variable-in-dll
 // issue, and for anyone dependent on ONNX it will be defined as
-// ONNX_IMPORT. ONNX_BUILD_MAIN_LIB can also be set when being built
-// statically if ONNX is being linked into a shared library that wants
-// to export the ONNX APIs and classes.
-//
+// ONNX_IMPORT.
 // More details on Windows dllimport / dllexport can be found at
 // https://msdn.microsoft.com/en-us/library/3y1sfaz2.aspx
 //
 // This solution is similar to
 // https://github.com/pytorch/pytorch/blob/master/caffe2/core/common.h
-#if defined(ONNX_BUILD_SHARED_LIBS) || defined(ONNX_BUILD_MAIN_LIB)
+
+#if defined(_MSC_VER)
+// dllimport will fail non-dll build, so extra flags is used.
+#if defined(ONNX_EXPORT_SYMBOLS)
 #define ONNX_API ONNX_EXPORT
```

Comment:
this is also not needed? Anyway, let's clean up firstly, and we may add it back if needed later :)